src/bvals/bvals_swarm.cpp

```diff
@@ -41,7 +41,7 @@ namespace parthenon {
 BoundarySwarm::BoundarySwarm(std::weak_ptr<MeshBlock> pmb)
     : bswarm_index(), pmy_block(pmb), pmy_mesh_(pmb.lock()->pmy_mesh) {
 #ifdef MPI_PARALLEL
-  swarm_id_ = pmb.lock()->pbval->bvars_next_phys_id_;
+  swarm_id_ = pmb.lock()->pbval->AdvanceCounterPhysID(1);
```

Comment:
```suggestion
  swarm_id_ = pmb.lock()->pbswarm->AdvanceCounterPhysID(1);
```
Yes this is a bug, thanks for noticing this @pgrete!